File: src/main/webapp/src/app/models/lighting-action-universe.ts
import { LightingActionChannelValue } from "./lighting-action-channel-value";

export class LightingActionUniverse {
  universeUuid: string;
  channelValueList: LightingActionChannelValue[] = [];

  constructor(data?: any) {
    if (!data) {
      return;
    }

    this.universeUuid = data.universeUuid;

    if (data.channelValueList) {
      for (let channelValue of data.channelValueList) {
        this.channelValueList.push(
          new LightingActionChannelValue(channelValue)
        );
      }
    }
  }
}

File: src/main/java/com/ascargon/rocketshow/lighting/LightingActionUniverse.java
package com.ascargon.rocketshow.lighting;

import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
public class LightingActionUniverse {

    private String universeUuid;
    private List<LightingActionChannelValue> channelValueList = new ArrayList<>();

}
